analyzers/src/SonarAnalyzer.VisualBasic/Rules/UriShouldNotBeHardcoded.cs

```diff
@@ -21,38 +21,29 @@
 namespace SonarAnalyzer.Rules.VisualBasic
 {
     [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
-    public sealed class UriShouldNotBeHardcoded
-        : UriShouldNotBeHardcodedBase<ExpressionSyntax, LiteralExpressionSyntax,
-            SyntaxKind, BinaryExpressionSyntax, ArgumentSyntax, VariableDeclaratorSyntax>
+    public sealed class UriShouldNotBeHardcoded : UriShouldNotBeHardcodedBase<SyntaxKind, LiteralExpressionSyntax, ArgumentSyntax>
     {
-        private static readonly DiagnosticDescriptor Rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
-        protected override SyntaxKind StringLiteralSyntaxKind => SyntaxKind.StringLiteralExpression;
-
-        protected override SyntaxKind[] StringConcatenateExpressions =>
-            new[]
-            {
-                SyntaxKind.AddExpression,
-                SyntaxKind.ConcatenateExpression
-            };
-
+        protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
         protected override GeneratedCodeRecognizer GeneratedCodeRecognizer => VisualBasicGeneratedCodeRecognizer.Instance;
+        protected override SyntaxKind[] StringConcatenateExpressions => new[] { SyntaxKind.AddExpression, SyntaxKind.ConcatenateExpression };
+        protected override SyntaxKind[] InvocationOrObjectCreationKind => new[] { SyntaxKind.InvocationExpression, SyntaxKind.ObjectCreationExpression };
 
-        protected override string GetLiteralText(LiteralExpressionSyntax literalExpression) => literalExpression?.Token.ValueText;
-
-        protected override ExpressionSyntax GetLeftNode(BinaryExpressionSyntax binaryExpression) => binaryExpression.Left;
-
-        protected override ExpressionSyntax GetRightNode(BinaryExpressionSyntax binaryExpression) => binaryExpression.Right;
+        protected override string GetLiteralText(LiteralExpressionSyntax literalExpression) =>
+            literalExpression?.Token.ValueText;
```

Comment:
C# and VB.NET implementations are exactly the same, except for the namespace of `LiteralExpressionSyntax`. While the base class takes a `TLiteralExpressionSyntax`, I guess you 
can't define the implementation at base level, because the C# version and the VB.NET version of the class don't share a common ancestor having `Token` defined.
If that's the case, what about extending the language facade, and in particular the `SyntaxFacade`?
The method can give back a generic `SyntaxToken` or even just a string corresponding to `ValueText`. We already have a method doing that for an expression:
```cs
    public string GetName(SyntaxNode expression) =>
        expression.GetName();
```

Reply:
As we talked about, this will be done in a subsequent PR.